packages/workers-shared/asset-server-worker/src/index.ts

```diff
@@ -1,38 +1,66 @@
+import { WorkerEntrypoint } from "cloudflare:workers";
 import { AssetsManifest } from "./assets-manifest";
-import { NoopAssetsManifest } from "./assets-manifest-no-op";
+import {
+	InternalServerErrorResponse,
+	MethodNotAllowedResponse,
+	NotFoundResponse,
+	OkResponse,
+} from "./responses";
+import { getAdditionalHeaders, getMergedHeaders } from "./utils/headers";
+import { getAssetWithMetadataFromKV } from "./utils/kv";
 
-interface Env {
-	ASSETS_MANIFEST: ArrayBuffer;
-	ASSETS_KV_NAMESPACE: KVNamespace;
-}
+export default class extends WorkerEntrypoint<Env> {
+	async fetch(request: Request) {
+		if (!request.method.match(/^(get)$/i)) {
+			return new MethodNotAllowedResponse();
+		}
 
-export default {
-	async fetch(request: Request, env: Env) {
-		const {
-			// ASSETS_MANIFEST is a pipeline binding to an ArrayBuffer containing the
-			// binary-encoded site manifest
-			ASSETS_MANIFEST = new ArrayBuffer(0),
+		try {
+			return this.handleRequest(request);
+		} catch (err) {
+			return new InternalServerErrorResponse(err);
+		}
+	}
 
-			// ASSETS_KV_NAMESPACE is a pipeline binding to the KV namespace that the
-			// assets are in.
-			ASSETS_KV_NAMESPACE,
-		} = env;
+	async hasResponseForRequest(request: Request) {
+		return !!(await this.getAssetEntry(request));
+	}
 
-		const url = new URL(request.url);
-		const { pathname } = url;
+	async handleRequest(request: Request) {
+		const assetEntry = await this.getAssetEntry(request);
+		if (!assetEntry) {
+			return new NotFoundResponse();
+		}
 
-		const isLocalDevContext = new Uint8Array(ASSETS_MANIFEST).at(0) === 1;
-		const assetsManifest = isLocalDevContext
-			? new NoopAssetsManifest()
-			: new AssetsManifest(ASSETS_MANIFEST);
-		const assetEntry = await assetsManifest.get(pathname);
+		const assetResponse = await getAssetWithMetadataFromKV(
+			this.env.ASSETS_KV_NAMESPACE,
+			assetEntry
+		);
+		console.log(assetResponse);
+		if (!assetResponse || !assetResponse.value) {
```

Comment:
It feels like `assetResponse.value` being falsy is a programming error on our part?

Reply:
technically this should be just a sanity check, but felt like we should check nonetheless. I'll ask the EWC folks if we want to throw a different error here